modulus/datapipes/climate/era5_hdf5.py

```diff
@@ -433,14 +460,23 @@ def __call__(self, sample_info: dali.types.SampleInfo) -> Tuple[Tensor, Tensor]:
         in_idx = idx % self.num_samples_per_year
 
         data = self.data_files[year_idx]["fields"]
-        # Has [C,H,W] shape.
-        invar = data[in_idx, self.chans]
+        if self.num_history == 0:
+            # Has [C,H,W] shape.
+            invar = data[in_idx, self.chans]
+        else:
+            # Has [T,C,H,W] shape.
+            invar = data[
+                in_idx : in_idx + (self.num_history + 1) * self.stride : self.stride,
+                self.chans,
+            ]
 
         # Has [T,C,H,W] shape.
-        outvar = np.empty((self.num_steps,) + invar.shape, dtype=invar.dtype)
+        outvar = np.empty(
```

Comment:
So `invar` here should already have `self.chans` channels, so can you do something like this:
```py
outvar = np.empty((self.num_steps,) + invar.shape[-3:], dtype=invar.dtype)
```
?

Reply:
Yes, switched to this